simln-lib/src/test_utils.rs

```diff
@@ -45,3 +54,187 @@ pub fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
+    }
+
+    /// Creates a new builder instance with the specified number of nodes.
+    /// The default configuration includes a balance of 100,000 units per node,
+    /// no keysend support, and no specific networks.
+    pub fn new(node_count: usize) -> Self {
+        Self {
+            node_count,
+            initial_balance: 100_000,    // Default 100k sats
+            keysend_indices: Vec::new(), // No keysend by default
+            networks: None,              // No specific networks by default
+        }
+    }
+
+    /// Specifies which nodes should have the keysend feature enabled
+    pub fn with_keysend_nodes(mut self, indices: Vec<usize>) -> Self {
+        self.keysend_indices = indices;
+        self
+    }
+
+    /// Sets specific networks for each node
+    /// Will panic if the number of networks doesn't match node_count
+    /// Returns self for method chaining
+    pub fn with_networks(mut self, networks: Vec<Network>) -> Self {
+        // Validate that we have the correct number of networks
+        if networks.len() != self.node_count {
+            panic!("Must specify a network for each node");
+        }
+        self.networks = Some(networks);
+        self
+    }
+
+    /// Builds only the client map, omitting node info.
+    /// Useful for network-specific testing. Returns a map of public keys to mocked
+    /// Lightning node clients.
+    pub fn build_clients_only(self) -> HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>> {
+        let (_, clients) = self.build_full();
+        clients
+    }
+
+    /// Builds the full test setup, including node info and clients.
+    /// Returns a tuple of node information and a map of public keys to mocked
+    /// Lightning node clients.
+    pub fn build_full(self) -> TestNodesResult {
+        let nodes = create_nodes(self.node_count, self.initial_balance);
+        let mut node_infos = Vec::new();
+        let mut clients: HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>> = HashMap::new();
+
+        for (idx, (mut node_info, _)) in nodes.into_iter().enumerate() {
+            if self.keysend_indices.contains(&idx) {
+                node_info.features.set_keysend_optional();
+            }
+
+            let mut mock_node = MockLightningNode::new();
+            mock_node.expect_get_info().return_const(node_info.clone());
+
+            if let Some(networks) = &self.networks {
+                let network = networks[idx];
+                mock_node
+                    .expect_get_network()
+                    .returning(move || Ok(network));
+            }
+
+            clients.insert(node_info.pubkey, Arc::new(Mutex::new(mock_node)));
+            node_infos.push(node_info);
+        }
+
+        (node_infos, clients)
+    }
+}
+
+pub fn create_simulation(
+    clients: HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
+    activity: Vec<ActivityDefinition>,
+) -> Simulation {
+    Simulation::new(
+        SimulationCfg::new(Some(0), 0, 0.0, None, None),
```

Comment:
Add a comment that this sets a runtime for the simulation of 0, so `run` will exit immediately - this probably isn't desirable for other tests. 

Happy to leave as-is for now, if we need it we can make it a parameter or add a builder.